### ISSUE TITLE
Add validation annotations

### ISSUE THREAD

**Issue by mib4fun** [jaumard/trailpack-annotations#26]
Hello jaumard,

Would be perfect to have validation annotations.
How can i do that ?

**Comment by jaumard** [comment]
You already have it ;) 

This should work (not tested) :  
```
/**
   * Return some info about this application
   * @Route({method: ["GET"], path: "/default/info", config: validation: {......}}) 
   */
  info (request, reply) {
    reply.json(this.app.services.DefaultService.getApplicationInfo())
  }
```

Let me know :)

**Comment by mib4fun** [comment]
Yeah, let me test that

**Comment by mib4fun** [comment]
The basic config is working good 
```
@Route({method: 'GET', path: '/auth/me',  config: {validate: { query: { a: false } }}}) 
```

But can't use something more complex with Joi like
```
   * @Route({method: 'GET', path: '/auth/me',  config: {validate: { query: Joi.object({ 'test': Joi.string().required() }).options({   allowUnknown: true  })}}}) 
```

I got this  
````
2:18:42 PM web.1 |  error: 
2:18:42 PM web.1 |   ReferenceError: Joi is not defined
2:18:42 PM web.1 |      at eval (eval at <anonymous> (/home/jekas/dev/lang/nodejs/trails/app/backend-ibs/node_modules/ecmas-annotations/lib/attribute-parser.js:162:16), <anonymous>:1:70)
2:18:42 PM web.1 |      at AttributeParser.<anonymous> (/home/jekas/dev/lang/nodejs/trails/app/backend-ibs/node_modules/ecmas-annotations/lib/attribute-parser.js:162:11)
2:18:42 PM web.1 |      at Array.forEach (native)
2:18:42 PM web.1 |      at AttributeParser.parse (/home/jekas/dev/lang/nodejs/trails/app/backend-ibs/node_modules/ecmas-annotations/lib/attribute-parser.js:125:11)
2:18:42 PM web.1 |      at Reader.parseAnnotations (/home/jekas/dev/lang/nodejs/trails/app/backend-ibs/node_modules/ecmas-annotations/lib/reader.js:253:45)
2:18:42 PM web.1 |      at Reader.parseMethodAnnotations (/home/jekas/dev/lang/nodejs/trails/app/backend-ibs/node_modules/ecmas-annotations/lib/reader.js:124:17)
2:18:42 PM web.1 |      at Reader.<anonymous> (/home/jekas/dev/lang/nodejs/trails/app/backend-ibs/node_modules/ecmas-annotations/lib/reader.js:74:39)
2:18:42 PM web.1 |      at Array.forEach (native)
2:18:42 PM web.1 |      at Reader.parse (/home/jekas/dev/lang/nodejs/trails/app/backend-ibs/node_modules/ecmas-annotations/lib/reader.js:73:15)
2:18:42 PM web.1 |      at _.each.iterate (/home/jekas/dev/lang/nodejs/trails/app/backend-ibs/node_modules/trailpack-annotations/lib/configure.js:40:12)
2:18:42 PM web.1 |      at Array.forEach (native)
2:18:42 PM web.1 |      at module.exports.annotations (/home/jekas/dev/lang/nodejs/trails/app/backend-ibs/node_modules/trailpack-annotations/lib/configure.js:55:9)
2:18:42 PM web.1 |      at AnnotationsTrailpack.configure (/home/jekas/dev/lang/nodejs/trails/app/backend-ibs/node_modules/trailpack-annotations/index.js:17:5)
2:18:42 PM web.1 |      at /home/jekas/dev/lang/nodejs/trails/app/backend-ibs/node_modules/trails/lib/trailpack.js:63:26
2:18:42 PM web.1 |  error: The application did not boot successfully.
2:18:42 PM web.1 |  error: Try increasing the loglevel to "debug" to learn more
````

**Comment by jaumard** [comment]
Hum I see... Sound tricky to resolved :/ when I parse classes to retrieve annotations I have all field as string and try to parse it as object when I can. So of course Joi is not define at this time because it was done under the hood by `ecmas-annotations`...

**Comment by mib4fun** [comment]
Yup, i got this working by defining Joi in this file ```ecmas-annotations/lib/attribute-parser.js```
Hmm, maybe we can overload the class ?

**Comment by jaumard** [comment]
Maybe there something more clean to do... Here I use `eval` to retrieve object (https://github.com/jaumard/ecmas-annotations/blob/master/lib/attribute-parser.js#L156) but instead VM API can be used (https://nodejs.org/api/vm.html) don't know if it's available since node V4 but if it's the case it will be more easy to fix by adding a globalAttribut array field to the library that can be use under the VM. The trailpack will pass `joi` and normally all will work as expected I think. 

Doing an overload will force us to modify `ecmas-anotation` deeply to allow this. What do you think  ?

**Comment by mib4fun** [comment]
WoW didn't know about VM API. Seems like the way to go !

**Comment by jaumard** [comment]
@mib4fun Just found it after some research about your error lol :D can you take a look at this ? I don't know when I will have the time for this. :/

**Comment by mib4fun** [comment]
Yeah, i got you there buddy ;)
How should i start ?

**Comment by jaumard** [comment]
Thanks! First step remove eval for VM :)

**Comment by mib4fun** [comment]
Ok ! it's failing quite nice :D . Starting debug

**Comment by mib4fun** [comment]
Just landed https://github.com/jaumard/ecmas-annotations/pull/9

**Comment by jaumard** [comment]
Fixed on v1.0.2 :) let you write tests ^^
